src/witan/sen2/return/person_level/blade/pre_submission.clj

```diff
@@ -0,0 +1,711 @@
+(ns witan.sen2.return.person-level.blade.pre-submission
+  "Read SEN2 Blade from pore-submission Excel workbook."
+  (:require [clojure.set :as set]
+            [clojure.string :as str]
+            [tech.v3.libs.poi :as poi]
+            [tech.v3.dataset.io.spreadsheet :as ss]
+            [tablecloth.api :as tc])
+  (:import [java.time LocalDate]
+           [java.time LocalDateTime]
+           [java.time.format DateTimeFormatter]))
+
+
+
+;;; # Utility functions
+(defn- parse-id
+  "Parse numeric ID (which may be read from Excel worksheet as float) `x` back to string representations of integer."
+  [x]
+  (cond
+    (float?  x) (format "%.0f" x)
+    (int?    x) (str x)
+    (string? x) (str/trim x)))
+
+(defn- parse-binary->boolean
+  "Parse 0 & 1 `x` (read from Excel worksheet as string or floats) to boolean."
+  [x]
+  (cond
+    (float?   x) (case (int x)  0 false 1 true nil)
+    (string?  x) (case      x  "0" false "1" true nil)
+    (boolean? x) x))
```

Comment:
worth having an `:else` that throws an `ex-info` with a bit of detail on what went wrong?